source/javascripts/components/StepConfiguration/components/StepInput/StepInput.tsx

```diff
@@ -34,21 +35,49 @@ function isTextareaInput(props: Props): props is TextareaProps {
 }
 
 const StepInput = forwardRef<Props, "textarea" | "select">((props: Props, ref) => {
-	const { label, isRequired, isSensitive, helperSummary, helperDetails, ...rest } = props;
+	const { label, isRequired, isSensitive, helperSummary, helperDetails, name = "", ...rest } = props;
 
-	const { watch, setValue } = useFormContext();
+	const { watch, setValue, getValues } = useFormContext();
 	const { open: openSecretsDialog } = useSecretsDialog();
+	const { open: openEnvironmentVariablesDialog } = useEnvironmentVariablesDialog();
+	const [cursorPosition, setCursorPosition] = useState<{ start: number; end: number }>();
 	const textareaRef = useAutosize<HTMLTextAreaElement>(ref);
 
-	const onClear = () => setValue(rest.name!, "");
-	const isClearableInput = isSensitive && !!watch(rest.name || "", props.defaultValue);
+	const onClear = () => setValue(name, "");
+	const isClearableInput = isSensitive && !!watch(name, props.defaultValue);
 
 	const handleOnClickInsertSecret: MouseEventHandler<HTMLButtonElement> = (e) => {
 		// NOTE: This is necessary because without it, the tooltip on the button reappears after the dialog is closed.
 		e.currentTarget.blur();
 
 		openSecretsDialog({
-			onSelect: (secret) => setValue(rest.name!, `$${secret.key}`),
+			onSelect: (secret) => setValue(name, `$${secret.key}`),
+		});
+	};
+
+	const handleOnClickInsertEnvironmentVariable: MouseEventHandler<HTMLButtonElement> = (e) => {
+		// NOTE: This is necessary because without it, the tooltip on the button reappears after the dialog is closed.
+		e.currentTarget.blur();
+
+		openEnvironmentVariablesDialog({
+			onSelect: ({ key }) => {
+				const value = getValues(name) as string;
+				const { start, end } = cursorPosition ?? { start: value.length, end: value.length };
+
+				setCursorPosition({ start, end: end + `$${key}`.length });
+				setValue(name, `${value.slice(0, start)}$${key}${value.slice(end)}`);
```

Comment:
Can we extract the cursor handling somehow, so it could be reused elsewhere too?

Reply:
Good idea, I will do!